lib/i18n_digests.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class I18nDigests
+  class << self
+    def build_digests(available_locales)
+      available_locales.each do |locale|
+        (self.i18n_digests ||= {})[locale.to_sym] = locale_file_digest(locale)
+      end
+    end
+
+    def for_locale(locale)
+      i18n_digests[locale.to_sym]
+    end
+
+    private
+
+    def i18n_digests
+      Rails.application.config.x.i18n_digests
```

Comment:
Why the app config? :thinking: I guess Rails may reload that for us when a locale changes. Is that true? Maybe we should mention that in a comment. Otherwise I would have been tempted to suggest a class instance var but that wouldn't update in dev.

Reply:
The idea behind using a key in app config for storing these values is that we initialize it once when the app starts (including on deployment of a new release) but before anything else loads (just after we've determined `I18n.available_locales`), and then it's available anywhere in the app in any context, eg: models, views, background jobs, services.

> I guess Rails may reload that for us when a locale changes. Is that true?

I'm not quite sure what you mean, but these values won't change when the user switches from one locale to another, it'll be set once during app startup. If the contents of any of the locale files (eg: `es.yml`) have changed, the digests will be different. The hash stored in `Rails.application.config.x.i18n_digests` will look something like this, assuming `I18n.available_locales` returns `["en", "es"]`:

```ruby
{
  en: "ed98f7a98f7a9ef7",
  es: "9ed8fa9d8f7a98aa",
}
```
If the contents of `config/locales/en.yml` and `config/locales/es.yml` don't change, then any time the app starts the values for those digests will stay exactly the same. If the content of `config/locales/es.yml` changes and the app starts, then the digest for `:es` will be different. So that means any cache keys that used `en` will be kept (cache hits) and any for `es` will be automatically invalidated (cache miss) and re-rendered and re-cached.

Does that make sense?

Reply:
> Does that make sense?

Yes, that all makes sense. I was just wondering if the config is reloaded during development when a locale changes. When classes change then Zeitwerk reloads them and there was a change we did in initializers to call them wrapped in `Rails.application.config.after_initialize`. I guessed that this code would re-run when classes are reloaded. But that was just a guess.

Anyway, we will see if this caching becomes stale during development and then we can solve that problem later.